### PR TITLE
OKTA-631946 - Run CI on CircleCI

### PR DESCRIPTION
**Pull request by moisesolmedo-okta**
### Problem Analysis (Technical)

Run CI on CCI

### Steps to reproduce:

Actual result:

Expected result:

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -2,44 +2,112 @@
...
+          paths:
+            - Logs/Test
+  functional-tests:
+    macos:
+      xcode: 14.3.1
+    resource_class: macos.m1.medium.gen1
     steps:
+      - checkout
+      - attach_workspace:
+          at: ./FT/DD
```
Comment (IldarAbdullin-okta) [inline .circleci/config.yml:68]: ?

File: .circleci/config.yml [hunk 2]
```diff
@@ -2,44 +2,112 @@
...
       - run:
-          name: "package stage"
-          command: "echo package stage"
-# Invoke jobs via workflows
-# See: https://circleci.com/docs/2.0/configuration-reference/#workflows
+          name: Functional tests
+          command: set -o pipefail && xcodebuild -workspace DeviceAuthenticator.xcworkspace -scheme "DeviceAuthenticatorFramework" -destination "platform=iOS Simulator,OS=latest,name=iPhone 14" -derivedDataPath FT/DD test-without-building -only-testing:DeviceAuthenticatorFunctionalTests -enableCodeCoverage YES | xcpretty
+      - persist_to_workspace:
+          root: FT
```
Comment (IldarAbdullin-okta) [inline .circleci/config.yml:73]: ?

File: .circleci/config.yml [hunk 3]
```diff
@@ -2,44 +2,112 @@
...
+          paths:
+            - DD/Logs/Test
```
Comment (IldarAbdullin-okta) [inline .circleci/config.yml:75]: DD == DerivedData?

**Review by IldarAbdullin-okta — APPROVED** [review]
LGTM!

File: .circleci/config.yml (hunk 1 above)
Comment (moisesolmedo-okta) [reply .circleci/config.yml]: FT == FunctionalTests but I removed this since it was for the code coverage step.

**Pull request merged**